### ISSUE TITLE
2.0.1 - PutMappingDescriptor.AutoMap ignores maxRecursion argument

### ISSUE THREAD

**Issue by hmanner** [elastic/elasticsearch-net#1825]
I have a recursive type mapping which I create using AutoMap(..., 10). This works fine when creating the mapping during index creation via the following method:
```
TypeMappingDescriptor<T> AutoMap(IPropertyVisitor visitor = null, int maxRecursion = 0)
```
However, I also need to update the mapping dynamically (on all levels) with new fields, which I perform via this method (different class):
```
PutMappingDescriptor<T> AutoMap(IPropertyVisitor visitor = null, int maxRecursion = 0)
```
This only updates the mapping on the top level. Looks like this method ignores the maxRecursion argument, it always sends 0 to the PropertyWalker.

In my case this is a critical issue, as I am using dynamic = strict. This will cause later attempts to set the new field a few levels down in the hierarchy of a document to fail, because the type mapping contains the new field only on top level.

**Comment by gmarz** [comment]
Hey @hmanner this is a dupe of https://github.com/elastic/elasticsearch-net/issues/1821 which we've already pushed a fix for.

We'll get a 2.0.2 release out asap.

**Comment by gmarz** [comment]
FYI @hmanner just released 2.0.2

**Comment by hmanner** [comment]
That was fast :) Sorry I missed that other issue before filing the duplicate.
